AnaProd/HH_bbWW/baseline.py

```diff
@@ -39,22 +39,25 @@ def RequestOnlyResolvedGenJets(df):
 
 
 def RecoHWWCandidateSelection(df):
-    df = df.Define("Electron_sel", """
-        v_ops::pt(Electron_p4) > 10 && abs(v_ops::eta(Electron_p4)) < 2.5 && abs(Electron_dz) < 0.2 && abs(Electron_dxy) < 0.045
-        && Electron_mvaIso_WP80 """) # this cut tells that electron is not inside of a jet
+    df = df.Define("Electron_presel", """
+        v_ops::pt(Electron_p4) > 10 && abs(v_ops::eta(Electron_p4)) < 2.5 && abs(Electron_dz) < 0.1 && abs(Electron_dxy) < 0.05 && Electron_sip3d <= 8 && Electron_miniPFRelIso_all < 0.4 && Electron_mvaIso_WP90""")
 
-    df = df.Define("Muon_sel", """
-        v_ops::pt(Muon_p4) > 15 && abs(v_ops::eta(Muon_p4)) < 2.4 && abs(Muon_dz) < 0.2 && abs(Muon_dxy) < 0.045
-        && Muon_tightId && Muon_pfRelIso04_all < 0.15""")
+    df = df.Define("Muon_presel", """
+        v_ops::pt(Muon_p4) > 15 && abs(v_ops::eta(Muon_p4)) < 2.4 && abs(Muon_dz) < 0.1 && abs(Muon_dxy) < 0.05 && abs(Muon_dxy) < 0.05 && Muon_sip3d <= 8 && Muon_miniPFRelIso_all < 0.4 && Muon_looseId""")
+
+    df = df.Define("Electron_sel", f"""
+        Electron_presel && Electron_mvaIso_WP80 """)
+
+    df = df.Define("Muon_sel", f"""
+        Muon_presel && ( (Muon_tightId && Muon_mediumPromptId ) || (Muon_highPtId && Muon_mediumPromptId ))""")
 
     df = df.Define("N_sel_Ele", "Electron_pt[Electron_sel].size()")
     df = df.Define("N_sel_Mu", "Muon_pt[Muon_sel].size()")
     df = df.Define("N_sel_lep", "N_sel_Ele + N_sel_Mu")
-    df = df.Filter("N_sel_lep > 0 && N_sel_lep < 3", "lepton selection")
 
     df = df.Define("is_SL", "N_sel_lep == 1")
     df = df.Define("is_DL", "N_sel_lep == 2")
```

Comment:
With the looser selection, more than two leptons could survive the selection. We need to select the best pair (or signal lep).

Reply:
changed to >=2